src/macros.h

```diff
@@ -56,20 +56,20 @@
     Nan::ThrowError(#var " must be a number"); \
     return; \
   } \
-  unsigned int var = name->Uint32Value();
+  int64_t var = name->IntegerValue();
 
 #define ASSERT_UINT_BOUNDS(name, var, min, max) \
   if (!name->IsNumber()) { \
     Nan::ThrowError(#var " must be a number"); \
     return; \
   } \
-  unsigned int var = name->Uint32Value(); \
+  int64_t var = name->IntegerValue(); \
   \
-  if (min > 0 && var < min) { \
+  if (var < min) { \
     Nan::ThrowError(#var " must be at least " #min); \
     return; \
   } \
-  if (max <= sizeof(unsigned int) && var > max) { \
+  if ((uint64_t) var > max) { \
```

Comment:
Yes, I was worried about this too. Maybe make sure that `min >= 0` would suffice? We're somewhat currently relying on the user of the macro knowing what's going on inside it